### PR TITLE
PXP-9373: Unit test for diff manifest

### PR DESCRIPTION
**Pull request by MaribelleHGomez**
Jira Ticket: [PXP-9373](https://ctds-planx.atlassian.net/browse/PXP-9373)

### New Features
- unit test added for the manifest differential functionality

### Deployment changes

[PXP-9373]: https://ctds-planx.atlassian.net/browse/PXP-9373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: gen3/tools/diff.py [hunk 1]
```diff
@@ -108,46 +109,53 @@ def _precheck_manifests(
...
+
+                manifest_content.append(content)
+
+        return {
+            "csvdict": manifest_content,
+            "headers": [x for x in headers[0] if x in headers[1]],
+        }
+
+    except Exception as e:
+        print(e)
```
Comment (BinamB) [inline gen3/tools/diff.py:158]: maybe you should return e?

File: tests/test_diff.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        },
+    )
+
+
+def test_file_input_mismatch():
+    """
+    Test for fail due to different file types.
+    """
+
+    with pytest.raises(SystemExit):
```
Comment (BinamB) [inline tests/test_diff.py:81]: I suppose I dont get this. I think it should be an assertion. Something that looks like 
```
assert manifest_diff(...) == error
```

File: tests/test_diff.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        manifest_diff(
+            files=[
+                "tests/test_data/manifest1.csv",
+                "tests/test_data/diff_manifests/manifest3.tsv",
+            ],
+        )
+
```
Comment (BinamB) [inline tests/test_diff.py:88]: I think maybe add one more test that tests the same file and shows that there are no differences between the files.

File: tests/test_diff.py (hunk 2 above)
Comment (MaribelleHGomez) [reply tests/test_diff.py:81]: If the files are two different types it exits and stops, what kind of error were you thinking?

Comment (frankliuao) [reply tests/test_diff.py:81]: My $0.02:
I'd thought that Gen3 would have some custom Exceptions but if not, would a `ValueError` be more appropriate than `TypeError`? The arguments for files might both be in strings? It's the read-in file contents that don't match. My naive thoughts.

**Comment by frankliuao** [comment]
Following up on this @BinamB @MaribelleHGomez

**Review by BinamB — APPROVED** [review]
lgtm! 🚀

**Pull request merged**